src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1318,74 +1319,83 @@ because xbuild doesn't support framework reference assemblies.
 	</ResolveLibraryProjectImports>
 </Target>
 
-<Target Name="_CollectLibraryResourceDirectories">
+<Target Name="_CollectLibraryResourceDirectories"
+		Condition="'$(_AndroidUseAapt2)' == 'True'"
+	>
 	<CollectNonEmptyDirectories Directories="@(LibraryResourceDirectories);@(_AdditonalAndroidResourceCachePaths->'%(Identity)\res')">
 		<Output TaskParameter="Output" ItemName="_LibraryResourceDirectories" />
 	</CollectNonEmptyDirectories>
+	<ComputeHash Source="@(_LibraryResourceDirectories)"	>
+		<Output TaskParameter="Output" ItemName="_LibraryResourceHashDirectories" />
+	</ComputeHash>
 </Target>
 
 <Target Name="_CompileAndroidLibraryResources" DependsOnTargets="_CollectLibraryResourceDirectories"
-		Condition="'$(_AndroidUseAapt2)' == 'True' And '$(DesignTimeBuild)' != 'True' "
-		Inputs="%(_LibraryResourceDirectories.FileFound)"
-		Outputs="%(_LibraryResourceDirectories.Identity)\..\compiled.stamp"
+		Condition="'$(_AndroidUseAapt2)' == 'True'"
+		Inputs="%(_LibraryResourceHashDirectories.FileFound)"
+		Outputs="$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).stamp"
 	>
+	<MakeDir Directories="$(_AndroidLibraryFlatArchivesDirectory)" Condition="!Exists('$(_AndroidLibraryFlatArchivesDirectory)')" />
 	<ConvertResourcesCases
 		Condition=" '@(_LibraryResourceDirectories)' != '' "
 		ContinueOnError="$(DesignTimeBuild)"
 		ResourceDirectories="@(_LibraryResourceDirectories)"
 		ResourceNameCaseMap="$(_AndroidResourceNameCaseMap)"
 		AcwMapFile="$(_AcwMapFile)"
 		CustomViewMapFile="$(_CustomViewMapFile)"
-		AndroidConversionFlagFile="%(_LibraryResourceDirectories.Identity)\..\compiled.flata"
+		AndroidConversionFlagFile="$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).stamp"
 	/>
 	<Aapt2Compile
-		Condition=" '@(_LibraryResourceDirectories)' != '' "
+		Condition=" '@(_LibraryResourceHashDirectories)' != '' "
 		ContinueOnError="$(DesignTimeBuild)"
-		ResourceDirectories="@(_LibraryResourceDirectories)"
+		ResourceDirectories="@(_LibraryResourceHashDirectories)"
 		ExplicitCrunch="$(AndroidExplicitCrunch)"
 		ExtraArgs="$(AndroidAapt2CompileExtraArgs)"
+		FlatArchivesDirectory="$(_AndroidLibraryFlatArchivesDirectory)"
 		ToolPath="$(Aapt2ToolPath)"
 		ToolExe="$(Aapt2ToolExe)"
 	/>
 	<Touch
-			Files="@(_LibraryResourceDirectories->'%(Identity)\..\compiled.stamp')"
+			Files="$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).stamp"
 			AlwaysCreate="True"
 	/>
 	<ItemGroup>
-		<FileWrites Include="@(_LibraryResourceDirectories.Identity->'%(Identity)\..\compiled.flata')"
-				Condition="Exists ('@(_LibraryResourceDirectories.Identity->'%(Identity)\..\compiled.flata')')"
+		<FileWrites Include="$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).flata"
+				Condition="Exists ('$(_AndroidLibraryFlatArchivesDirectory)%(_LibraryResourceHashDirectories.Hash).flata')"
```

Comment:
Is there really a case where the `*.flata` files won't exist? Can we just remove the `Condition`?